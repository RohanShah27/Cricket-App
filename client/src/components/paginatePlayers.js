import React, { Component } from "react";
import "../styles/Players.css";
import virat from "./rishabh.jpg";
import { getAllPlayers, playerSearch } from "../actions/Players";
import { connect } from "react-redux";
import InfiniteScroll from "react-infinite-scroll-component";
import { Link } from "react-router-dom";

export class paginatePlayers extends Component {
  constructor(props) {
    super(props);
  }
  state = {
    playerName: "",
    items: 10,
    loadingState: false,
    height: 800,
    pagePlayers: []
  };
  componentDidMount() {
    this.props.getAllPlayers();
  }
  componentWillReceiveProps(nextProps) {
    nextProps.players.length > 0
      ? this.displayPlayers(nextProps.players)
      : console.log(0, " players");
  }
  displayPlayers = players => {
    const { items } = this.state;

    if (players.length === 0) return;
    let pagePlayers = [];
    for (let i = 0; i < items; i++) {
      pagePlayers.push(players[i]);
    }
    this.setState({ pagePlayers });
    console.log("PagePlayers array", this.state.pagePlayers.length);
    // return this.pagePlayers;
  };
  genMarkup = player => (
    <div className="players-card">
      <img src={virat} className="players-img "></img>
      <p className="player-name">{player.player_name}</p>
      <span className="team-name">
        <span style={{ color: "black" }}>Team:</span>{" "}
        {player.nation ? player.nation : "NA"}
      </span>
      <button
        onClick={() =>
          this.props.history.push("/playerprofile/" + player.player_id, {
            player
          })
        }
      >
        View
      </button>
    </div>
  );
  loadMoreItems = () => {
    // this.setState({ loadingState: true, height: this.state.height + 800 });
    setTimeout(() => {
      this.setState({
        items: this.state.items + 10
        //   loadingState: false
      });
      this.displayPlayers(this.props.players);
    }, 1000);
  };
  searchForPlayer = () => {
    let playerName = {
      playerName: this.state.playerName
    };
    this.props.playerSearch(playerName);
  };
  OnChange = event => {
    this.setState({ [event.target.name]: event.target.value });
    let playerName = {
      playerName: this.state.playerName
    };
    this.props.playerSearch(playerName);
  };
  render() {
    return (
      <div style={{ marginTop: "80px" }}>
        <div className="players-search-section">
          <input
            type="text"
            placeholder="Search for Players"
            name="playerName"
            value={this.state.playerName}
            onChange={this.OnChange}
          />
          <button onClick={this.searchForPlayer}>
            {" "}
            <i class="fa fa-search"></i>
          </button>
          {this.state.playerName == "" ? null : (
            <div className="player-search-result">
              <span className="search_result_data">
                <p>
                  {this.props.search_result
                    ? this.props.search_result.map(player => (
                        <Link
                          style={{
                            textDecoration: "none"
                          }}
                          to={{
                            pathname: "/playerprofile/" + player.player_id,
                            state: { player }
                          }}
                        >
                          <p
                            style={{
                              fontSize: "18px",
                              paddingTop: "2px ",
                              display: "flex",
                              justifyContent: "center"
                            }}
                          >
                            {player.player_name}
                          </p>
                        </Link>
                      ))
                    : null}
                </p>
              </span>
            </div>
          )}
        </div>
        <InfiniteScroll
          dataLength={this.state.pagePlayers.length} //This is important field to render the next data
          next={this.loadMoreItems}
          hasMore={true}
          height={600}
          loader={
            <div className="loader-container">
              <div className="user-loader"></div>
            </div>
          }
          endMessage={
            <p style={{ textAlign: "center" }}>
              <b>Yay! You have seen all Players</b>
            </p>
          }
        >
          <div className="players-card-container">
            {this.state.pagePlayers.map(player => {
              return (
                <div className="players-card">
                  <img src={virat} className="players-img "></img>
                  <p className="player-name">{player.player_name}</p>
                  <span className="team-name">
                    <span style={{ color: "black" }}>Team:</span>{" "}
                    {player.nation ? player.nation : "NA"}
                  </span>
                  <button
                    onClick={() =>
                      this.props.history.push(
                        "/playerprofile/" + player.player_id,
                        {
                          player
                        }
                      )
                    }
                  >
                    View
                  </button>
                </div>
              );
            })}
          </div>
        </InfiniteScroll>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  players: state.playersReducer.players,
  search_result: state.playersReducer.search_player
});
export default connect(
  mapStateToProps,
  { getAllPlayers, playerSearch }
)(paginatePlayers);
