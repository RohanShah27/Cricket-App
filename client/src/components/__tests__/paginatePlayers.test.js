import React from "react";
import { shallow, mount } from "enzyme";
import { PaginatePlayers } from "../paginatePlayers";
import { BrowserRouter as Router } from "react-router-dom";
const history = { push: jest.fn() };
const paginate = jest.fn();
const getAllPlayers = jest.fn();
const playerSearch = jest.fn();
const searchForPlayer = jest.fn();
let pagePlayers = [];
for (var i = 1; i < 21; i++) {
  pagePlayers.push({
    player_id: i,
    player_name: "test name",
    player_nation: "test nation"
  });
}
let players = [
  {
    player_id: 1,
    player_name: "test name",
    player_nation: "test nation"
  }
];
let search_result = [
  {
    player_id: 1,
    player_name: "virat"
  },
  {
    player_id: 2,
    player_name: "Kohli"
  }
];
const wrapper = mount(
  <Router>
    <PaginatePlayers
      paginate={paginate}
      playerSearch={playerSearch}
      getAllPlayers={getAllPlayers}
      history={history}
      search_result={search_result}
      searchForPlayer={searchForPlayer}
    />
  </Router>
);

describe("Test paginatePlayers Component", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.state()).toBe("");
  });
  // it("should check for component did mount as a function", () => {
  //   const componentDidMount = jest.spyOn(
  //     PaginatePlayers.prototype,
  //     "componentDidMount"
  //   );
  //   wrapper.instance().componentDidMount();
  //   expect(componentDidMount).toHaveBeenCalled();
  //   expect(getAllPlayers).toHaveBeenCalled();
  // });
  // it("should check for componentWillReceiveProps mount as a function", () => {
  //   const componentWillReceiveProps = jest.spyOn(
  //     PaginatePlayers.prototype,
  //     "componentWillReceiveProps"
  //   );
  //   let playersArray = [];
  //   for (var i = 1; i < 21; i++) {
  //     playersArray.push({
  //       player_id: i,
  //       player_name: "test name",
  //       player_nation: "test nation"
  //     });
  //   }
  //   let nextProps = {
  //     players: playersArray
  //   };
  //   const displayPlayers = jest.spyOn(wrapper.instance(), "displayPlayers");
  //   wrapper.instance().componentWillReceiveProps(nextProps);
  //   expect(displayPlayers).toHaveBeenCalledWith(nextProps.players);
  // });
  // it("should dislay playerName", () => {
  //   // expect(wrapper.state().pagePlayers).toBe("");
  //   const displayPlayers = jest.spyOn(wrapper.instance(), "displayPlayers");
  //   expect(displayPlayers).toHaveBeenCalledWith(pagePlayers);
  //   expect(wrapper.find("#playerName0").props().children).toBe("test name");
  // });
  // it("should have only one search bar", () => {
  //   expect(wrapper.find("#searchPlayer").length).toBe(1);
  // });
  // it("should have placeholder", () => {
  //   expect(wrapper.find("#searchPlayer").prop("placeholder")).toBe(
  //     "Search for Players"
  //   );
  // });
  // it("shoudl check for input value", () => {
  //   const e = {
  //     target: {
  //       name: "playerName",
  //       value: "virat"
  //     }
  //   };
  //   expect(wrapper.state().playerName).toBe("");
  //   wrapper.instance().OnChange(e);
  //   expect(wrapper.state().playerName).toBe(e.target.value);
  //   expect(playerSearch).toBeCalledWith({ playerName: e.target.value });
  // });
  // it("should search player on click of button", () => {
  //   expect(wrapper.find("#searchButton").simulate("click"));
  //   expect(searchForPlayer).toBeCalledTimes(0);
  // });
  it("should display search results", () => {
    wrapper.setState({ playerName: "Virat" });
    console.log(wrapper.props().props);
    expect(wrapper.find("#name0").text()).toBe(search_result[0].player_name);
  });
});
