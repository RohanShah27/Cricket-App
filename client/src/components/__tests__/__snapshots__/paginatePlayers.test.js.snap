// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test paginatePlayers Component should render the component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <PaginatePlayers
      getAllPlayers={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      paginate={[MockFunction]}
      playerSearch={[MockFunction]}
      searchForPlayer={[MockFunction]}
      search_result={
        Array [
          Object {
            "player_id": 1,
            "player_name": "virat",
          },
          Object {
            "player_id": 2,
            "player_name": "Kohli",
          },
        ]
      }
    >
      <div
        style={
          Object {
            "marginTop": "80px",
          }
        }
      >
        <div
          className="players-search-section"
        >
          <input
            id="searchPlayer"
            name="playerName"
            onChange={[Function]}
            placeholder="Search for Players"
            type="text"
            value=""
          />
          <button
            id="searchButton"
            onClick={[Function]}
          >
             
            <i
              class="fa fa-search"
            />
          </button>
        </div>
        <InfiniteScroll
          dataLength={0}
          disableBrowserPullToRefresh={true}
          endMessage={
            <p
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <b>
                Yay! You have seen all Players
              </b>
            </p>
          }
          hasMore={true}
          height={600}
          loader={
            <div
              className="loader-container"
            >
              <div
                className="user-loader"
              />
            </div>
          }
          next={[Function]}
          pullDownToRefreshContent={
            <h3>
              Pull down to refresh
            </h3>
          }
          pullDownToRefreshThreshold={100}
          releaseToRefreshContent={
            <h3>
              Release to refresh
            </h3>
          }
        >
          <div
            style={Object {}}
          >
            <div
              className="infinite-scroll-component "
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "height": 600,
                  "overflow": "auto",
                }
              }
            >
              <div
                className="players-card-container"
              />
              <div
                className="loader-container"
              >
                <div
                  className="user-loader"
                />
              </div>
            </div>
          </div>
        </InfiniteScroll>
      </div>
    </PaginatePlayers>
  </Router>
</BrowserRouter>
`;
